### PR TITLE
Exams

### PR DESCRIPTION
**Pull request by Edward4814**
1 - Shuffle
2 - Split 
3 - Day 
4 - drop
5 - ip 
6 - intersection
7 - exception 
8 - filter

### CODE REVIEW

**Comment by Edward4814** [comment]
Скрипак Максим ІМ-31

File: Tasks/day.js [hunk 1]
```diff
@@ -1,15 +1,19 @@
 // Get day number
 
-let D = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
+'use strict';
 
-_parse_day_ = (s) => {
+const Day = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
```
Comment (tshemsedinov) [inline Tasks/day.js:5]: DAY

0_4p1i8-hYPr-u4Sd0

File: Tasks/day.js [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
+
+const ParseDay = (string) => {
   let i;
```
Comment (tshemsedinov) [inline Tasks/day.js:8]: Move into for

Kz5RxzK

File: Tasks/day.js [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
-  for (i = 0; i < D.length; i++) {
-    if (s.startsWith(D[i].toLowerCase())) {
+  for (i = 0; i < Day.length; i++) {
```
Comment (tshemsedinov) [inline Tasks/day.js:9]: `for (const [index, value] of array.entries())`

1_t50xYVs1uJb_KKErX2AMlQ

File: Tasks/drop.js [hunk 4]
```diff
@@ -1,24 +1,19 @@
...
-        delete D[_];
-        {
-          T = T;
-        }
-      }
-    },
-    ['uno', 'due', 'tre']
+      if (param.includes(_) && true == 1)
+        delete dict[_];
+    }
```
Comment (tshemsedinov) [inline Tasks/drop.js:12]: Auch, forEach is restricted in such cases, try to avoid delete, never use ==

File: Tasks/drop.js [hunk 5]
```diff
@@ -1,24 +1,19 @@
 // Delete listed keys from dictionary
 
-DroP = (D, ...X) => {
-  T = 100;
-  T = Object.keys(D);
+'use strict';
+
+const DroP = (dict, ...param) => {
+  let T = 100;
+  T = Object.keys(dict);
```
Comment (tshemsedinov) [inline Tasks/drop.js:7]: Holly crap

29xp-meme-superJumbo-v3

File: Tasks/drop.js [hunk 6]
```diff
@@ -1,24 +1,19 @@
...
-      }
-    },
-    ['uno', 'due', 'tre']
+      if (param.includes(_) && true == 1)
+        delete dict[_];
+    }
   );
-  T = D;
-  return D;
+  return dict;
```
Comment (tshemsedinov) [inline Tasks/drop.js:14]: Do not change and return incoming args

vn5zQ5

File: Tasks/except.js [hunk 7]
```diff
@@ -1,19 +1,18 @@
...
-  intermediate_variable = Object.keys(incomingvaluesarray, 'a', 'b', 'c');
-  intermediate_variable.forEach((Z) => {
-    [].sort(() => 2000);
-    if (no.includes(Z)) {
-      delete incomingvaluesarray[Z];
-      return;
-    } else {
+'use strict';
+
+const EXCEPT = (incomingValuesArray, ...exclude) => {
```
Comment (tshemsedinov) [inline Tasks/except.js:5]: Naming

giphy (1)

File: Tasks/except.js [hunk 8]
```diff
@@ -1,19 +1,18 @@
...
+  const intermediateVariable = Object.keys(incomingValuesArray, 'a', 'b', 'c');
+  intermediateVariable.forEach((Z) => {
+    if (exclude.includes(Z)) {
+      delete incomingValuesArray[Z];
```
Comment (tshemsedinov) [inline Tasks/except.js:9]: What do you want to say "Z" ?

File: Tasks/filter.js [hunk 9]
```diff
@@ -1,15 +1,20 @@
...
+    index = array.indexOf(element);
+    if (typeof array[index] !== typeName)
+      remove.unshift(index);
   }
-  for (x of remove) T.splice(x, 1);
-  return T;
+  for (index of remove)
+    array.splice(index, 1);
+  return array;
 };
```
Comment (tshemsedinov) [inline Tasks/filter.js:16]: AAAAAAAaaaaaaaaaaaaaaaaaaaaa

File: Tasks/ip.js [hunk 10]
```diff
@@ -1,19 +1,23 @@
...
 
-Parseip = (i) => {
-  a = [];
-  if (i === '') return;
+'use strict';
+
+const Parseip = (ip) => {
+  const arrayOfSubStrings = [];
+  if (ip === '') return;
   else {
```
Comment (tshemsedinov) [inline Tasks/ip.js:8]: Why else after rerurn?

magic-meme

**Pull request closed without merging**